packages/menu/src/useMenu.ts

```diff
@@ -39,25 +39,32 @@ import {
 } from './types';
 
 export const useMenu = <T extends HTMLElement = HTMLElement, M extends HTMLElement = HTMLElement>({
-  items: rawItems,
-  idPrefix,
+  defaultExpanded = false,
+  defaultFocusedValue,
+  document: documentProp,
   environment,
-  menuRef,
-  triggerRef,
-  rtl = false,
-  onChange = () => undefined,
+  focusedValue,
+  idPrefix,
   isExpanded,
-  defaultExpanded = false,
+  items: rawItems,
+  menuRef,
   restoreFocus = true,
+  rtl = false,
   selectedItems,
-  focusedValue,
-  defaultFocusedValue
+  triggerRef,
+  window: windowProp,
+  onChange = () => undefined
 }: IUseMenuProps<T, M>): IUseMenuReturnValue => {
   const prefix = `${useId(idPrefix)}-`;
   const triggerId = `${prefix}menu-trigger`;
   const isExpandedControlled = isExpanded !== undefined;
   const isSelectedItemsControlled = selectedItems !== undefined;
   const isFocusedValueControlled = focusedValue !== undefined;
+
+  const _window = windowProp || environment;
+  let _document: Document | ShadowRoot = _window ? _window.document : window.document;
+  _document = documentProp ? documentProp : _document;
```

Comment:
```suggestion
 let _document: IUseMenuProps['document'];
 
 if (documentProp) {
   _document = documentProp
 } else {
   _document = _window ? _window.document : window.document;
 }
```

...for improved readability and type reuse. Repeated ternaries are difficult to read and usually a signal for refactor.